src/lib/route-segments.js

```diff
@@ -0,0 +1,42 @@
+export const STOPS = [[15, '3px'], [17, '4px'], [18, '10px'], [20, '45px']]
+export const ZERO_SPEED_STOPS = [[15, '0.5px'], [17, '1px'], [18, '2.5px'], [20, '11.25px']]
+
+// Calculate the slope and y-intercept in order to get linear equation
+// Values given as params are (x, y) and (c, d)
+export function getLinearValue (x, y, c, d, zoom) {
+  // Calculating slope
+  const m = (y - d) / (x - c)
+  // Calculating y-intercept
+  const b = y - (m * x)
+  // Getting linear interpolation of zoom value
+  const value = (m * zoom) + b
+  return value
+}
+
+// Replicating how tangram "stops" data structure works
+export function getSegmentWidth (zoom, speed) {
+  // If speed is zero, use zeroSpeed stops for less weight
+  const array = (speed === 0 || speed === null || typeof speed === 'undefined') ? ZERO_SPEED_STOPS : STOPS
+  const startValue = array[0]
+  const endValue = array[array.length - 1]
+  // If zoom values are outside the defined range,
+  // then they are capped by highest and lowest values in range
+  if (zoom < startValue[0]) {
+    return Number(startValue[1].slice(0, -2))
```

Comment:
I realized what is happening here - it may be simpler to say `window.parseInt(startValue[1], 10)` - it converts a value like `"3px"` to `3` in one step.

Reply:
I didn't use `parseInt` because I realized that some of the values in `ZERO_SPEED_STOPS` are decimal values and if I remember correctly, `parseInt()` will change it to an integer? Does that matter? I guess it doesn't if it's pretty close, just wasn't sure.

Reply:
Oh, you're right -- I didn't notice the decimal pixel values. I don't know enough about Tangram to know if decimal pixel values mean anything, but to get around it we could substitute `parseInt` with `parseFloat` which would preserve the decimals if they exist. Would that affect anything else?

Reply:
oh I didn't even know that existed. Okay I will that. Nope, I don't think that will affect anything else.